components/Search.tsx

```diff
@@ -33,17 +33,32 @@ const StyledInputBase = styled((props: InputBaseProps) => <OutlinedInput {...pro
 }))
 
 const Search = () => {
-  const { push, asPath } = useRouter()
+  const { push, asPath, query } = useRouter()
   const searchRef = useRef<HTMLInputElement | null>(null)
   const [showClearBtn, setShowClearBtn] = useState(false)
   const isHome = asPath === '/' || asPath === '/zh-CN'
   const theme = useTheme()
 
+  useEffect(() => {
+    if (!searchRef.current) return
+    if (query.search) {
+      searchRef.current.value = (query.search as string) || ''
+    } else {
+      // on page refresh, next.js's query.search will always be null, so need this fallback
+      const queryValue = new URLSearchParams(asPath.split('?')[1]).get('search')
```

Comment:
Actually, the `asPath.split('?')[1]` could be replaced by `window.location.search`